Data/ByteString/Lazy.hs

```diff
@@ -525,6 +526,85 @@ minimum (Chunk c cs) = foldlChunks (\n c' -> n `min` S.minimum c')
                                      (S.minimum c) cs
 {-# INLINE minimum #-}
 
+-- | /O(c)/ 'compareLength' compares the length of a 'ByteString' 
+-- to an 'Int64'   
+compareLength :: ByteString -> Int64 -> Ordering
+compareLength _ toCmp | toCmp < 0 = GT
+compareLength Empty toCmp         = compare 0 toCmp
+compareLength (Chunk c cs) toCmp  = compareLength cs (toCmp - fromIntegral (S.length c))
+{-# INLINE compareLength #-}
+
+{-# RULES 
+"ByteString.Lazy length/compareN -> compareLength" [~1] forall t n.
+  compare (length t) n = compareLength t n
+  #-}
+
+{-# RULES 
+"ByteString.Lazy compareN/length -> compareLength" [~1] forall t n.
+  compare n (length t) = toEnum $ (2-) $ fromEnum $ compareLength t n
```

Comment:
I agree that the `fromEnum`-`toEnum` bit is a somewhat hard to read.

How about using a helper defined with pattern matching instead?

```haskell
negateOrdering :: Ordering -> Ordering
negateOrdering LT = GT
negateOrdering EQ = EQ
negateOrdering GT = LT
```